version: '3.3'

services:

  vault:
    image: docker.io/vault:latest
    environment:
      - VAULT_LOCAL_CONFIG=${VAULT_LOCAL_CONFIG?Variable not set}
    networks:
      - vault-network
      - traefik-network
    deploy:
      placement:
        constraints:
          - node.role == manager
          - node.platform.os == linux
      labels:
        - traefik.enable=true
        - traefik.docker.network=traefik-network
        - traefik.constraint-label=traefik-proxy
        - traefik.http.routers.vault-http.rule=Host(`vault.${DOMAIN?Variable not set}`)
        - traefik.http.routers.vault-http.entrypoints=http
        - traefik.http.routers.vault-http.middlewares=https-redirect
        - traefik.http.routers.vault-https.rule=Host(`vault.${DOMAIN?Variable not set}`)
        - traefik.http.routers.vault-https.entrypoints=https
        - traefik.http.routers.vault-https.tls=true
        - traefik.http.routers.vault-https.tls.certresolver=le
        - traefik.http.services.vault.loadbalancer.server.port=8200

networks:
  vault-network:
    attachable: true
  traefik-network:
    external: true
